monitor/__init__.py

```diff
@@ -0,0 +1,213 @@
+    if res.status_code != 200:
+        raise ValueError(
+            'Request to slack returned an error %s, the response is:\n%s'
+            % (res.status_code, res.text)
+        )
+
+
+def format_data_size(value: int):
+    base = 1024
+    value = float(value)
+    suffix = ('kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB')
+    if value < base:
+        return '%d Bytes' % value
+    for i, s in enumerate(suffix):
+        unit = base ** (i + 2)
+        if value < unit:
+            return f'{round((base * value / unit),2)} {s}'
+
+
+def search_cloudwatch_dates(start_time: datetime, end_time: datetime, group_name: str, filter_pattern: str):
+    # this function is used to figure out what logstreams to look inside, if we dont do this, boto3 searches through all
+    # log streams starting from the lambda initialization, this can lead to hitting limits for python recursion.
+    log_stream_lookup = list()
+    log_stream_lookup.append(f"{end_time.strftime('%Y/%m/%d')}/")
+    log_stream_lookup.append(f"{(end_time - datetime.timedelta(days=1)).strftime('%Y/%m/%d')}/")
+    events = list()
+    for logs_stream_prefix in log_stream_lookup:
+        new_events = get_cloudwatch_log_events(start_time, end_time, group_name, filter_pattern, logs_stream_prefix)
+        events.extend(new_events)
+    return events
+
+
+def get_cloudwatch_log_events(start_time: datetime, end_time: datetime, group_name: str, filter_pattern: str,
+                              log_stream_prefix: str):
+        paginator = logsmanager.get_paginator('filter_log_events')
+        epoch = datetime.datetime.utcfromtimestamp(0)
+        events = []
+
+        kwargs = {'endTime': int((end_time - epoch).total_seconds()*1000),
+                  'startTime': int((start_time - epoch).total_seconds()*1000),
+                  'logGroupName': group_name, 'filterPattern': filter_pattern, 'interleaved': True,
+                  'logStreamNamePrefix': log_stream_prefix}
+        for page in paginator.paginate(**kwargs):
+            if len(page["events"]) > 0:
+                for x in page["events"]:
+                    index = x['message'].find('{')
+                    x = json.loads(x['message'][index:])
+                    events.append(x)
+
+        return events
+
+
+def run(webhook: bool = None):
+    if os.environ["DSS_INFRA_TAG_STAGE"] is None:
+        raise ValueError('Missing DSS_INFRA_TAG_STAGE, exiting....')
+        exit(1)
+
+    # variables
+    aws_end_time = datetime.datetime.utcnow()
+    aws_start_time = aws_end_time - datetime.timedelta(days=1)
+    bucket_list = [os.environ['DSS_S3_BUCKET'], os.environ['DSS_S3_CHECKOUT_BUCKET']]
+    bucket_query_metric_names = ['BytesDownloaded', 'BytesUploaded']
+    lambda_query_metric_names = ['Duration', 'Invocations']
+    notification_filter_type = ['DELETE', 'CREATE', 'TOMBSTONE']
+    stages = {f'{os.environ["DSS_INFRA_TAG_STAGE"]}': collections.defaultdict(collections.defaultdict)}
```

Comment:
I'm confused by the presence of multiple stages in this method.

Does the `dev` version of DSS-Monitor track multiple DSS deployments?

Reply:
Originally, i wanted to have 1 lambda that would perform all of the work across each HCA-Stages (excluding prod). 
I think that this can still be possible (perhaps preferable), it would just require modification to how the deployment script sets the Swagger URL for the HCA CLI. 

this naming convention was a residual from that.